### PR TITLE
feat(client): use ng-templates to generate templateCache

### PR DESCRIPTION
**Pull request by thnkloud9**
this reduces the amount of XHR requests on initial page load, as well as fixing issues with loading unwrapped templateUrls (urls not wrapped with asset.templateUrl()) when using superdesk client as a bower dependency.

Please let me know if you see any problems with this method, or have any suggestions or concerns.

### CODE REVIEW

**Comment by plamut** [comment]
This is the kind of stuff we need, I like the optimization!

:+1:

File: client/Gruntfile.js [hunk 1]
```diff
@@ -16,7 +16,26 @@ module.exports = function (grunt) {
         tasksDir: 'tasks',
         bowerDir: 'bower',
         poDir: 'po',
-        livereloadPort: 35729
+        livereloadPort: 35729,
+        ngtemplates: {
```
Comment (petrjasek) [inline client/Gruntfile.js]: you can put this into `tasks/options/ngtemplates.js` it will pick it from there.

File: client/Gruntfile.js [hunk 2]
```diff
@@ -27,6 +46,9 @@ module.exports = function (grunt) {
         configPath: require('path').join(process.cwd(), 'tasks', 'options')
     });
 
+    // automatically minify and cache HTML templates into $templateCache
+    grunt.loadNpmTasks('grunt-angular-templates');
```
Comment (petrjasek) [inline client/Gruntfile.js]: i think this is not needed, `load-grunt-tasks` should take care of this

**Comment by petrjasek** [comment]
looks good
would this also work for you @nistormihai @vied12 ?

**Comment by vied12** [comment]
:white_check_mark: Yeah ! I can't wait to try it :ok_hand:

**Comment by thnkloud9** [comment]
any chance to merge this?

**Comment by amagdas** [comment]
it has to pass e2e first...

**Comment by thnkloud9** [comment]
hmm.  no idea why these are failing.  and it seems to fail with different output each time.  any help here would be very much appreciated.

**Comment by thnkloud9** [comment]
also, strange that e2e tests are passing for my local repo:

https://travis-ci.org/thnkloud9/superdesk/builds/77355068

**Comment by plamut** [comment]
Our e2e tests are flaky and unstable, unfortunately, they very often report false errors :/

**Comment by ioanpocol** [comment]
I will take a look at the e2e errors and I will fix them

**Pull request closed without merging**